qiime2/sdk/tests/test_bad_inputs.py

```diff
@@ -0,0 +1,161 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import collections
+import tempfile
+import qiime2.core.archive as archive
+
+from qiime2.core.testing.util import get_dummy_plugin
+from qiime2.plugin.testing import TestPluginBase
+
+from qiime2.sdk import Artifact, Visualization
+from qiime2.core.testing.type import IntSequence1, IntSequence2, SingleInt
+from qiime2.core.testing.visualizer import most_common_viz
+from qiime2 import Metadata
+from qiime2.metadata.tests.test_io import get_data_path
+
+
+class TestBadInputs(TestPluginBase):
+
+    def make_provenance_capture(self):
+        # Copy-pasted from qiime2.sdk.TestResult.make_provenance_capture.
```

Comment:
Since this is a within-project copy-paste I don't think you need this comment. You could always just re-write the original comment to say something like "importing visualizations is not supported, but we do that here to simplify testing machinery".

Reply:
🗸